service_container/service_subscribers.rst

```diff
@@ -77,15 +77,178 @@ However, injecting the entire container is discouraged because it gives too
 broad access to existing services and it hides the actual dependencies of the
 services.
 
-**Service Locators** are intended to solve this problem by giving access to a
-set of predefined services while instantiating them only when actually needed.
+**Service Subscribers** are intended to solve this problem by giving access to a
+set of predefined services while instantiating them only when actually needed
+through a **Service Locator**, a separate lazy-loaded container.
+
+Defining a Service Subscriber
+-----------------------------
+
+First, turn ``CommandBus`` into an implementation of :class:`Symfony\\Component\\DependencyInjection\\ServiceSubscriberInterface`.
+Use its ``getSubscribedServices`` method to include as many services as needed
+in the service subscriber and change the type hint of the container to
+a PSR-11 ``ContainerInterface``::
+
+    // src/AppBundle/CommandBus.php
+    namespace AppBundle;
+
+    use AppBundle\CommandHandler\BarHandler;
+    use AppBundle\CommandHandler\FooHandler;
+    use Psr\Container\ContainerInterface;
+    use Symfony\Component\DependencyInjection\ServiceSubscriberInterface;
+
+    class CommandBus implements ServiceSubscriberInterface
+    {
+        private $locator;
+
+        public function __construct(ContainerInterface $locator)
+        {
+            $this->locator = $locator;
+        }
+
+        public static function getSubscribedServices()
+        {
+            return [
+                'AppBundle\FooCommand' => FooHandler::class,
+                'AppBundle\BarCommand' => BarHandler::class
+            ];
+        }
+
+        public function handle(Command $command)
+        {
+            $commandClass = get_class($command);
+
+            if ($this->locator->has($commandClass)) {
+                $handler = $this->locator->get($commandClass);
+
+                return $handler->handle($command);
+            }
+        }
+    }
+
+.. tip::
+
+    If the container does *not* contain the subscribed services, double-check
+    that you have :ref:`autoconfigure <services-autoconfigure>` enabled. You
+    can also manually add the ``container.service_subscriber`` tag.
+
+The injected service is an instance of :class:`Symfony\\Component\\DependencyInjection\\ServiceLocator`
+which implements the PSR-11 ``ContainerInterface``, but it is also a callable::
+
+    // ...
+    $locateHandler = $this->locator;
+    $handler = $locateHandler($commandClass);
+
+    return $handler->handle($command);
+
+Including services
```

Comment:
`services` -> `Services` because we use a special case for titles. Same in the rest of titles of the article.